### PR TITLE
Update var to let/const

### PR DESCRIPTION
**Pull request by adeobas**
#### Summary
Updated page to replace "var" with "let" or "const" 

#### Motivation

It is bad practice to use var. 

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md [hunk 1]
```diff
@@ -11,8 +11,6 @@ tags:
   - speech
   - synthesis
 ---
-{{DefaultAPISidebar("Web Speech API")}}
```
Comment (queengooborg) [inline files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:14]: Accidental removal?

File: files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md [hunk 2]
```diff
@@ -58,18 +56,18 @@ Let's look at the JavaScript in a bit more detail.
 As mentioned earlier, Chrome currently supports speech recognition with prefixed properties, therefore at the start of our code we include these lines to feed the right objects to Chrome, and any future implementations that might support the features without a prefix:
 
 ```js
-var SpeechRecognition = window.SpeechRecognition || webkitSpeechRecognition;
-var SpeechGrammarList = window.SpeechGrammarList || webkitSpeechGrammarList;
-var SpeechRecognitionEvent = window.SpeechRecognitionEvent || webkitSpeechRecognitionEvent;
+let SpeechRecognition = window.SpeechRecognition || webkitSpeechRecognition;
+let SpeechGrammarList = window.SpeechGrammarList || webkitSpeechGrammarList;
+let SpeechRecognitionEvent = window.SpeechRecognitionEvent || webkitSpeechRecognitionEvent;
```
Comment (queengooborg) [inline files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:61]:
```suggestion
const SpeechRecognition = window.SpeechRecognition || webkitSpeechRecognition;
const SpeechGrammarList = window.SpeechGrammarList || webkitSpeechGrammarList;
const SpeechRecognitionEvent = window.SpeechRecognitionEvent || webkitSpeechRecognitionEvent;
```

File: files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md [hunk 3]
```diff
@@ -58,18 +56,18 @@ Let's look at the JavaScript in a bit more detail.
...
 
 #### The grammar
 
 The next part of our code defines the grammar we want our app to recognize. The following variable is defined to hold our grammar:
 
 ```js
-var colors = [ 'aqua' , 'azure' , 'beige', 'bisque', 'black', 'blue', 'brown', 'chocolate', 'coral' ... ];
-var grammar = '#JSGF V1.0; grammar colors; public <color> = ' + colors.join(' | ') + ' ;'
+let colors = [ 'aqua' , 'azure' , 'beige', 'bisque', 'black', 'blue', 'brown', 'chocolate', 'coral' ... ];
+let grammar = '#JSGF V1.0; grammar colors; public <color> = ' + colors.join(' | ') + ' ;'
```
Comment (queengooborg) [inline files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:70]:
```suggestion
const colors = [ 'aqua' , 'azure' , 'beige', 'bisque', 'black', 'blue', 'brown', 'chocolate', 'coral' ... ];
const grammar = '#JSGF V1.0; grammar colors; public <color> = ' + colors.join(' | ') + ' ;'
```

File: files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md [hunk 4]
```diff
@@ -84,8 +82,8 @@ The grammar format used is [JSpeech Grammar Format](https://www.w3.org/TR/jsgf/)
 The next thing to do is define a speech recognition instance to control the recognition for our application. This is done using the {{domxref("SpeechRecognition.SpeechRecognition()","SpeechRecognition()")}} constructor. We also create a new speech grammar list to contain our grammar, using the {{domxref("SpeechGrammarList.SpeechGrammarList()","SpeechGrammarList()")}} constructor.
 
 ```js
-var recognition = new SpeechRecognition();
-var speechRecognitionList = new SpeechGrammarList();
+let recognition = new SpeechRecognition();
+let speechRecognitionList = new SpeechGrammarList();
```
Comment (queengooborg) [inline files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:86]:
```suggestion
const recognition = new SpeechRecognition();
const speechRecognitionList = new SpeechGrammarList();
```

File: files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md [hunk 5]
```diff
@@ -114,11 +112,11 @@ recognition.maxAlternatives = 1;
 After grabbing references to the output {{htmlelement("div")}} and the HTML element (so we can output diagnostic messages and update the app background color later on), we implement an onclick handler so that when the screen is tapped/clicked, the speech recognition service will start. This is achieved by calling {{domxref("SpeechRecognition.start()")}}. The `forEach()` method is used to output colored indicators showing what colors to try saying.
 
 ```js
-var diagnostic = document.querySelector('.output');
-var bg = document.querySelector('html');
-var hints = document.querySelector('.hints');
+let diagnostic = document.querySelector('.output');
+let bg = document.querySelector('html');
+let hints = document.querySelector('.hints');
```
Comment (queengooborg) [inline files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:117]:
```suggestion
const diagnostic = document.querySelector('.output');
const bg = document.querySelector('html');
const hints = document.querySelector('.hints');
```

File: files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md [hunk 6]
```diff
@@ -137,7 +135,7 @@ Once the speech recognition is started, there are many event handlers that can b
 
 ```js
 recognition.onresult = function(event) {
-  var color = event.results[0][0].transcript;
+  let color = event.results[0][0].transcript;
```
Comment (queengooborg) [inline files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:138]:
```suggestion
  const color = event.results[0][0].transcript;
```

File: files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md [hunk 7]
```diff
@@ -230,18 +228,18 @@ Let's investigate the JavaScript that powers this app.
 First of all, we capture references to all the DOM elements involved in the UI, but more interestingly, we capture a reference to {{domxref("Window.speechSynthesis")}}. This is API's entry point — it returns an instance of {{domxref("SpeechSynthesis")}}, the controller interface for web speech synthesis.
 
 ```js
-var synth = window.speechSynthesis;
+let synth = window.speechSynthesis;
```
Comment (queengooborg) [inline files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:231]:
```suggestion
const synth = window.speechSynthesis;
```

File: files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md [hunk 8]
```diff
@@ -230,18 +228,18 @@ Let's investigate the JavaScript that powers this app.
...
 
-var inputForm = document.querySelector('form');
-var inputTxt = document.querySelector('.txt');
-var voiceSelect = document.querySelector('select');
+let inputForm = document.querySelector('form');
+let inputTxt = document.querySelector('.txt');
+let voiceSelect = document.querySelector('select');
```
Comment (queengooborg) [inline files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:235]:
```suggestion
const inputForm = document.querySelector('form');
const inputTxt = document.querySelector('.txt');
const voiceSelect = document.querySelector('select');
```

File: files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md [hunk 9]
```diff
@@ -230,18 +228,18 @@ Let's investigate the JavaScript that powers this app.
...
 
-var pitch = document.querySelector('#pitch');
-var pitchValue = document.querySelector('.pitch-value');
-var rate = document.querySelector('#rate');
-var rateValue = document.querySelector('.rate-value');
+let pitch = document.querySelector('#pitch');
+let pitchValue = document.querySelector('.pitch-value');
+let rate = document.querySelector('#rate');
+let rateValue = document.querySelector('.rate-value');
```
Comment (queengooborg) [inline files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:240]:
```suggestion
const pitch = document.querySelector('#pitch');
const pitchValue = document.querySelector('.pitch-value');
const rate = document.querySelector('#rate');
const rateValue = document.querySelector('.rate-value');
```

File: files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md [hunk 10]
```diff
@@ -230,18 +228,18 @@ Let's investigate the JavaScript that powers this app.
...
 
-var voices = [];
+let voices = [];
```
Comment (queengooborg) [inline files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:242]:
```suggestion
const voices = [];
```

File: files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md [hunk 11]
```diff
@@ -255,7 +253,7 @@ function populateVoiceList() {
   voices = synth.getVoices();
 
   for(i = 0; i < voices.length ; i++) {
-    var option = document.createElement('option');
+    let option = document.createElement('option');
```
Comment (queengooborg) [inline files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:256]:
```suggestion
    const option = document.createElement('option');
```

File: files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md [hunk 12]
```diff
@@ -290,8 +288,8 @@ Finally, we set the {{domxref("SpeechSynthesisUtterance.pitch")}} and {{domxref(
 inputForm.onsubmit = function(event) {
   event.preventDefault();
 
-  var utterThis = new SpeechSynthesisUtterance(inputTxt.value);
-  var selectedOption = voiceSelect.selectedOptions[0].getAttribute('data-name');
+  let utterThis = new SpeechSynthesisUtterance(inputTxt.value);
+  let selectedOption = voiceSelect.selectedOptions[0].getAttribute('data-name');
```
Comment (queengooborg) [inline files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:292]:
```suggestion
  const utterThis = new SpeechSynthesisUtterance(inputTxt.value);
  const selectedOption = voiceSelect.selectedOptions[0].getAttribute('data-name');
```

**Review by queengooborg — CHANGES_REQUESTED** [review]
Thank you for your PR!  I see a lot of places where `const` would be more applicable, but LGTM overall!

File: files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md (hunk 1 above)
Comment (adeobas) [reply files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:14]: Thank you for your suggestions. I reviewed them and approved the suggestions. 
I didn't mean to remove. I submitted a separate PR request to add the sidebar. I'm assuming that the update var to let/const removed it?

Comment (queengooborg) [reply files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:14]: No worries, it happens!  You can add it back in by going to `Files changed` > `...` (top-right corner) > `Edit file`!  (Apparently I can't make a suggestion here, don't know why...)

**Review by queengooborg — APPROVED** [review]
Thank you, LGTM!

**Pull request merged**